cli/compilers/ts.rs

```diff
@@ -217,14 +213,8 @@ impl TsCompiler {
 
     // If `checkJs` is set to true in `compilerOptions` then we're gonna be compiling
     // JavaScript files as well
-    let config_json = config.json()?;
-    let compile_js = match &config_json.get("compilerOptions") {
-      Some(serde_json::Value::Object(m)) => match m.get("checkJs") {
-        Some(serde_json::Value::Bool(bool_)) => *bool_,
-        _ => false,
-      },
-      _ => false,
-    };
+    let config_str = config.string()?;
+    let compile_js = CHECK_JS_RE.is_match(&config_str);
```

Comment:
Can you do a unit test here- showing comments in tsconfig rather than an integration test ?

Reply:
You're right :+1: rewritten as unit tests